Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/events/server/qrcodepostertemplate/QrCodePosterTemplateCache.kt

```diff
@@ -0,0 +1,41 @@
+package de.rki.coronawarnapp.eventregistration.events.server.qrcodepostertemplate
+
+import android.content.Context
+import de.rki.coronawarnapp.util.di.AppContext
+import timber.log.Timber
+import java.io.File
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class QrCodePosterTemplateCache @Inject constructor(@AppContext private val context: Context) {
```

Comment:
Why new class? Lukas already used Retrofit `Cache` to cache the response in app cache directory `context.filesDir `, I think we should also rely on the response headers. Please double check the specs?

Reply:
Yes, we could implement these caching cases by implementing an okhttp interceptor instead of managing caching ourself. 

I think it can be realised by adding an Interceptor that returns the cached response if we have no internet connection (like [here](https://stackoverflow.com/questions/23429046/can-retrofit-with-okhttp-use-cache-data-when-offline)) and extend it so that it also returns the cached response when the backend returns an unsuccessful response.

Reply:
yes, custom implementation was overkill, okhttp cache seems to work as expected out of the box. It can handle error states and offline mode as well.